core/dnsserver/onstartup.go

```diff
@@ -2,15 +2,27 @@ package dnsserver
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
 )
 
+// Below checkDomainName() function checks whether the given domain-name is following RFC1035 preferred syntax or not.
+// If not correct then shows a warning against domain-name.
+func checkDomainName(addr string) bool {
+	matchDomain, _ := regexp.Compile("[A-Za-z]+://([A-Za-z0-9-]+\\.[A-Za-z0-9-]+)*:[0-9]+")
+	return matchDomain.MatchString(addr)
+}
+
 // startUpZones creates the text that we show when starting up:
 // grpc://example.com.:1055
 // example.com.:1053 on 127.0.0.1
 func startUpZones(protocol, addr string, zones map[string]*Config) string {
 	s := ""
 
+	if !checkDomainName(addr) {
```

Comment:
`addr` is the wrong thing to validate here.  

Each zone key in `zones` needs to be validated.